### DISCUSSION TITLE
DefaultSettings class that can be overriden

### DISCUSSION THREAD

**Discussion by Andre601** [Chew/JDA-Chewtils#20] [Ideas]
When I understand the entirety of the command system correctly, is there no simple way to set certain defaults on a global scale.

Like there doesn't seem to be a way to define a bot's default Permissions required for all commands and you have to set the exact same permissions for each command separately.

This can become quite frustrating when you want to apply the same values on dozens of commands, as you now have to go and set it for each separate command class yourself.

Because of this would I like to suggest a way for devs to make a class, which implements/extends a `CommandSettings` class from Chewtils, which you can then set as the main default settings for the Command Handler/builder to use in commands, if they don't override things.

Example setup:
```java
// Not sure if it would be extends or implements
public class MyBotSettings extends CommandSettings {

    public boolean isGuildOnly() {
        return false; // Should also work on DM
    }
    
    public Permission[] getBotPermissions() {
        return new Permission[]{
            Permission.MESSAGE_WRITE,
            Permission.MESSAGE_EMBED_LINKS
        }
     }
} 
```

```java
CommandClientBuilder builder = new CommandClientBuilder();

builder.setDefaultSettings(new MyBotSettings());
```

**Comment by Chew** [comment]
I agree, there's a lot of options in the CommandClientBuilder and it's getting VERY messy.

Probably a 2.0 thing depending on how breaking we want to make it.

**Reply by Andre601** [reply]
I kinda want it asap, considering the message intent bullshit from discord...

**Reply by Chew** [reply]
Isn't message intent stuff handled by JDA?

**Reply by Andre601** [reply]
It's more about the fact that I now have to set the same bot perms 30+ times (once for each command) which is annoying

**Reply by Chew** [reply]
Oh, so you're wanting like, default command permissions?

**Reply by Andre601** [reply]
Just a general default settings thing imo...
